frontend/rolecall/src/app/cast/cast-drag-and-drop.component.ts

```diff
@@ -141,11 +141,13 @@ export class CastDragAndDrop implements OnInit {
     uuid: APITypes.CastUUID;
     saveDeleteEnabled?: boolean;
   }) {
-    this.buttonsEnabled = saveDeleteEnabled;
-    this.castSelected = true;
-    this.selectedCastUUID = uuid;
-    if (this.dataLoaded) {
-      this.setupData();
+    if (uuid) {
```

Comment:
We should never pass undefined in to this function. It would be better if the callers did the check and never called this function at all if uuid is undefined. 

But if it's really unavoidable then the input type must be changed to:
`uuid: APITypes.CastUUID | undefined;`

Reply:
I copied in a routine that calls selectCast. It is in a different component so it has less information about the inner workings of selectCast. As you can see, it seems to be standard procedure to set variables that has no meaningful value to 'undefined'. Still, sometimes '' is used (for this.urlUUID).

It is quite possible to make sure that nobody calls selectCast with a null uuid. However, I feel that it probably makes sense to put in a test inside selectCast just to be sure.

  setCurrentCast({cast, index}: {
    cast: Cast | undefined,
    index?: number,
  }) {
    if (index === undefined && cast) {
      index = this.selectedPieceCasts.findIndex(
          findCast => cast.uuid === findCast.uuid);
    }
    if (index >= 0) {
      this.lastSelectedCastIndex = index;
    } else {
      this.lastSelectedCastIndex = undefined;
    }
    this.lastSelectedCast = cast ? cast : this.lastSelectedCast;
    this.selectedCast = cast;
    this.dragAndDrop.selectCast({uuid: cast ? cast.uuid : undefined});
    this.urlUUID = cast ? cast.uuid : '';
    this.setCastURL();
  }

Reply:
When I was debugging the customer bug, selectCast was repeatedly called with an undefined uuid. I would like to put in a safeguard against bad things happening if selectCast is called with an undefined uuid.

I can create a new PR with only this change.

Reply:
Sure let's do a new clean PR with just that change and any related type changes if needed.